### PR TITLE
Pending QuickSend Correct Fee TEST

### PR DESCRIPTION
**Pull request by fluidvanadium**
Previously, fees were miscalculated before scan. FIXED

Whoohoo! This branch buffs the with_assertions module to make sure everything is right from the initial record, even before scanning the broadcast.
Previous incarnations of this branch failed. As recently as Wednesday, the following tests failed:
  
 --- STDERR:              libtonode-tests::chain_generics chain_generics::propose_and_broadcast_40_000_to_transparent ---
thread 'chain_generics::propose_and_broadcast_40_000_to_transparent' panicked at /home/cupress/ZINGO/libs/pending/zingo-testutils/src/assertions.rs:32:9:
assertion `left == right` failed
  left: 55000
 right: 15000

--- STDERR:              libtonode-tests::chain_generics chain_generics::send_shield_cycle ---
thread 'chain_generics::send_shield_cycle' panicked at /home/cupress/ZINGO/libs/pending/zingo-testutils/src/assertions.rs:32:9:
assertion `left == right` failed
  left: 124000
 right: 20000

However, merging todayz dev fixed this!

### CODE REVIEW

**Comment by zancas** [comment]
@fluidvanadium should you click "Update Branch"?

**Comment by zancas** [comment]
I clicked update branch.

File: zingo-testutils/src/lightclient/with_assertions.rs [hunk 1]
```diff
@@ -29,6 +29,7 @@ where
 
     environment.bump_chain().await;
 
+    assert_send_outputs_match_sender(client, &proposal, &txids).await;
```
Comment (zancas) [inline zingo-testutils/src/lightclient/with_assertions.rs]: I am having a hard time parsing this name.

To me it seems like this is in a mod called `assertions`.. so if I know which mod it lives in, then I know it's an `assertion`.

It seems like this code shows that the LRZ calculated:

`step.balance().fee_required()`

matches the:

`records.calculate_transaction_fee(record)`

Then if there's agreement, that `fee` is returned.

So, this is a fn in a mod name assertions but it does not return a bool.   That's my intuition for a fn in an assertions mod.

I'll propose-as-code.

**Review by zancas — DISMISSED** [review]

File: zingo-testutils/src/lightclient/with_assertions.rs [hunk 2]
```diff
@@ -69,10 +71,11 @@ where
     environment.bump_chain().await;
 
     recipient.do_sync(false).await.unwrap();
-    assert_send_outputs_match_recipient(recipient, &proposal, &txids).await;
+    get_proposal_vs_records_matched_total_output_value(recipient, &proposal, &txids).await;
```
Comment (fluidvanadium) [inline zingo-testutils/src/lightclient/with_assertions.rs]: actually, im afraid somebody will see this line does nothing, not realize it asserts, and delete it. it needs to start with 'assert

**Review by zancas — APPROVED** [review]

**Pull request merged**